src/libraries/System.Private.CoreLib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -368,84 +372,49 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
             }
 
             ref Entry entry = ref Unsafe.NullRef<Entry>();
-            if (_buckets != null)
```

Comment:
Still want an `if` test here (perhaps `.Length > 0` instead, or it will do a lot of work for an empty dictionary?